### PR TITLE
Sidebar: create single sidebar item component

### PR DESCRIPTION
**Pull request by mtias**
Continuing cleanup duties for the sidebar, this PR condenses the custom sidebar-item we had in My Sites and Me under `layout/sidebar/item.jsx`. The Reader is not using any component for this and will need some manual conversion.

The actual component could use a bit of cleanup itself, specially around the way it handles "selected" state.

### CODE REVIEW

File: client/me/sidebar/sidebar-item.jsx [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-
-		if ( this.props.onClick ) {
-			this.props.onClick( event );
-		}
-	},
-
-	render: function() {
-		const selected = this.props.selected ? 'selected' : null;
-		return (
-			<li className={ classnames( this.props.className, selected, 'me-sidebar-item' ) } >
```
Comment (aduth) [inline client/me/sidebar/sidebar-item.jsx]: There's a handful of `.me-sidebar-item` styles in `style.scss` that should probably be removed.

File: client/me/sidebar/sidebar-item.jsx [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
- */
-var layoutFocus = require( 'lib/layout-focus' ),
-	Gridicon = require( 'components/gridicon' );
-
-module.exports = React.createClass( {
-
-	displayName: 'MeSidebarItem',
-
-	onClick: function( event ) {
-		layoutFocus.setNext( 'content' );
```
Comment (aduth) [inline client/me/sidebar/sidebar-item.jsx]: Are we losing anything by not migrating this to `<SidebarItem />` ?

Comment (mtias) [inline client/me/sidebar/sidebar-item.jsx]: Probably. I forgot to move it to the sidebar wrapper.

Comment (mtias) [inline client/me/sidebar/sidebar-item.jsx]: Fixed.

**Pull request merged**